controller/src/main/java/io/pravega/controller/fault/ControllerClusterListener.java

```diff
@@ -67,6 +68,26 @@ public ControllerClusterListener(final Host host, final Cluster cluster,
         this.sweepers = Lists.newArrayList(sweepers);
     }
 
+    public boolean isZKConnected() {
+        if (cluster instanceof ClusterZKImpl) {
```

Comment:
is it possible to not use `instanceof` and add a method to `io.pravega.common.cluster.Cluster` like "isMetadataServiceConnected".
usually `instanceof` is a code smell in these cases

Also I won't call it "isZKConnected" but something that is not directly related to ZK, at this level it is not important that we are dealing with ZK, it is an implementation detail

Reply:
Fixed